### PR TITLE
git-team: init at 1.7.0

### PR DESCRIPTION
**Pull request by lockejan**
###### Description of changes

This PR adds git-team to nixpkgs.
It's a cli utility to easily add co-authors to commits.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/version-management/git-and-tools/git-team/default.nix
Comment (loicreynier) [inline pkgs/applications/version-management/git-and-tools/git-team/default.nix:19]:
```suggestion
      "Command line interface for managing and enhancing git commit messages with co-authors";
```

See [Submitting Changes](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes).

**Review by loicreynier — CHANGES_REQUESTED** [review]

File: pkgs/applications/version-management/git-and-tools/git-team/default.nix
Comment (ius) [inline pkgs/applications/version-management/git-and-tools/git-team/default.nix:18]: Minor style nitpick:
> Don’t include a period at the end. Don’t include newline characters. Capitalise the first character. For brevity, don’t repeat the name of package — just describe what it does.


https://nixos.org/manual/nixpkgs/stable/#var-meta-description

Comment (lockejan) [reply pkgs/applications/version-management/git-and-tools/git-team/default.nix:18]: Lesson learned.

File: pkgs/applications/version-management/git-and-tools/git-team/default.nix
Comment (lockejan) [reply pkgs/applications/version-management/git-and-tools/git-team/default.nix]: Lesson learned. :D

File: maintainers/maintainer-list.nix
Comment (zowoq) [inline maintainers/maintainer-list.nix:7238]: This should be a separate commit: `maintainers: add lockejan`

File: pkgs/applications/version-management/git-and-tools/git-team/default.nix
Comment (zowoq) [inline pkgs/applications/version-management/git-and-tools/git-team/default.nix:16]: This is a no-op with `vendorSha256 = null`
```suggestion
```

File: pkgs/applications/version-management/git-and-tools/git-team/default.nix
Comment (zowoq) [inline pkgs/applications/version-management/git-and-tools/git-team/default.nix:23]: Default for `buildGoModule`.
```suggestion
```

**Review by zowoq — CHANGES_REQUESTED** [review]
Needs to be listed in `pkgs/top-level/all-packages.nix` as well.

File: maintainers/maintainer-list.nix
Comment (lockejan) [reply maintainers/maintainer-list.nix:7238]: Makes sense. I someone guessed it is preferred to put it into one commit. At least I found a PR which hasn't had a distinct commit for that but I'm all for atomic commits. I probably just found a bad example.

File: pkgs/applications/version-management/git-and-tools/git-team/default.nix
Comment (lockejan) [reply pkgs/applications/version-management/git-and-tools/git-team/default.nix:16]: I don't think so. At least, building the package fails when omitting proxyVendor:
```
error: builder for '/nix/store/5zlpw3hqp966wq29qf487qhrcgc4157p-git-team-1.7.0.drv' failed with exit code 1;
       last 10 log lines:
       > configuring
       > building
       > Building subPackage .
       > go: inconsistent vendoring in /private/tmp/nix-build-git-team-1.7.0.drv-0/source:
       >        github.com/fatih/color@v1.11.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
       >        github.com/urfave/cli/v2@v2.3.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
       >
       >        To ignore the vendor directory, use -mod=readonly or -mod=mod.
       >        To sync the vendor directory, run:
       >                go mod vendor
       For full logs, run 'nix log /nix/store/5zlpw3hqp966wq29qf487qhrcgc4157p-git-team-1.7.0.drv'.
Traceback (most recent call last):
  File "/nix/store/q369ksyp8jr1ng9fjgb6vaj61wzkk4j4-nix-update-0.5.0/bin/.nix-update-wrapped", line 9, in <module>
    sys.exit(main())
  File "/nix/store/q369ksyp8jr1ng9fjgb6vaj61wzkk4j4-nix-update-0.5.0/lib/python3.9/site-packages/nix_update/__init__.py", line 199, in main
    nix_build(options)
  File "/nix/store/q369ksyp8jr1ng9fjgb6vaj61wzkk4j4-nix-update-0.5.0/lib/python3.9/site-packages/nix_update/__init__.py", line 153, in nix_build
    run(
  File "/nix/store/q369ksyp8jr1ng9fjgb6vaj61wzkk4j4-nix-update-0.5.0/lib/python3.9/site-packages/nix_update/utils.py", line 35, in run
    return subprocess.run(
  File "/nix/store/z740mapbgfi953b61jx72c0jy3phfcy8-python3-3.9.10/lib/python3.9/subprocess.py", line 528, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['nix', 'build', '--experimental-features', 'nix-command', '-f', './.', 'git-team']' returned non-zero exit status 1.
```

File: pkgs/applications/version-management/git-and-tools/git-team/default.nix
Comment (lockejan) [reply pkgs/applications/version-management/git-and-tools/git-team/default.nix:23]: Ok, that works. 
I followed the example given in the docs: https://nixos.org/manual/nixpkgs/stable/#ssec-language-go

Feels like that that part should be mentioned in there too.

File: pkgs/applications/version-management/git-and-tools/git-team/default.nix
Comment (zowoq) [inline pkgs/applications/version-management/git-and-tools/git-team/default.nix:16]: `proxyVendor` isn't needed.
```suggestion
  vendorSha256 = "sha256-xJMWPDuqoNtCCUnKuUvwlYztyrej1uZttC0NsDvYnXI=";
```

File: pkgs/applications/version-management/git-and-tools/git-team/default.nix
Comment (zowoq) [reply pkgs/applications/version-management/git-and-tools/git-team/default.nix:16]: This is actually a bug, I'll open a PR to correct it.

File: pkgs/applications/version-management/git-and-tools/git-team/default.nix
Comment (lockejan) [reply pkgs/applications/version-management/git-and-tools/git-team/default.nix:16]: How can I obtain this hash? Would come in handy to know for updates to come. I didn't find a good explanation yet.

File: pkgs/applications/version-management/git-and-tools/git-team/default.nix
Comment (zowoq) [reply pkgs/applications/version-management/git-and-tools/git-team/default.nix:16]: Only works if sandboxing is disabled (default on darwin), fails if it is enabled (default on linux).
```
nix-build -A git-team --option sandbox true
this derivation will be built:
  /nix/store/l4hbz1p5v7qb2jszdq1k7dvadiab0spg-git-team-1.7.0.drv
building '/nix/store/l4hbz1p5v7qb2jszdq1k7dvadiab0spg-git-team-1.7.0.drv'...
unpacking sources
unpacking source archive /nix/store/df3ry1f7dfvvda1xiyw72hc0l479z0rk-source
source root is source
patching sources
configuring
building
Building subPackage .
go: github.com/fatih/color@v1.11.0: Get "https://proxy.golang.org/github.com/fatih/color/@v/v1.11.0.mod": dial tcp: lookup proxy.golang.org: no such host
go: downloading github.com/urfave/cli/v2 v2.3.0
go: downloading github.com/fatih/color v1.11.0
go: github.com/fatih/color@v1.11.0: Get "https://proxy.golang.org/github.com/fatih/color/@v/v1.11.0.mod": dial tcp: lookup proxy.golang.org: no such host
error: builder for '/nix/store/l4hbz1p5v7qb2jszdq1k7dvadiab0spg-git-team-1.7.0.drv' failed with exit code 1;
       last 10 log lines:
       > unpacking source archive /nix/store/df3ry1f7dfvvda1xiyw72hc0l479z0rk-source
       > source root is source
       > patching sources
       > configuring
       > building
       > Building subPackage .
       > go: github.com/fatih/color@v1.11.0: Get "https://proxy.golang.org/github.com/fatih/color/@v/v1.11.0.mod": dial tcp: lookup proxy.golang.org: no such host
       > go: downloading github.com/urfave/cli/v2 v2.3.0
       > go: downloading github.com/fatih/color v1.11.0
       > go: github.com/fatih/color@v1.11.0: Get "https://proxy.golang.org/github.com/fatih/color/@v/v1.11.0.mod": dial tcp: lookup proxy.golang.org: no such host
       For full logs, run 'nix log /nix/store/l4hbz1p5v7qb2jszdq1k7dvadiab0spg-git-team-1.7.0.drv'.
```

File: pkgs/applications/version-management/git-and-tools/git-team/default.nix
Comment (lockejan) [reply pkgs/applications/version-management/git-and-tools/git-team/default.nix]: I updated default.nix accordingly and could just move on, but I don't just want to do stuff I don't fully understand. If you can give an explanation or point me into the right direction, that would be great.

Comment (zowoq) [reply pkgs/applications/version-management/git-and-tools/git-team/default.nix]: Not sure I understand what you're asking? https://github.com/NixOS/nixpkgs/pull/164372#discussion_r827926584 explains why the previous approach was incorrect?

Comment (lockejan) [reply pkgs/applications/version-management/git-and-tools/git-team/default.nix]: Well yes, but I still don't fully understand why vendorSha256 is needed to guarantee a reproducible build, because there is also the hash of the revision.🤔
I was hoping for an easy explanation or link why that is.

Comment (lockejan) [reply pkgs/applications/version-management/git-and-tools/git-team/default.nix]: I found some discussion in googlegroups. Fine for me now.

**Comment by lockejan** [comment]
Ah, shot. I forgot providing man-page and shell completion. Not ready yet. Sorry.

**Comment by lockejan** [comment]
Ready for re-review @zowoq @loicreynier (:

File: pkgs/top-level/all-packages.nix
Comment (ckiee) [inline pkgs/top-level/all-packages.nix:26144]:
```suggestion
  git-team = callPackage ../applications/version-management/git-and-tools/git-team { };
```

File: pkgs/applications/version-management/git-and-tools/git-team/default.nix
Comment (ckiee) [inline pkgs/applications/version-management/git-and-tools/git-team/default.nix:22]: Maybe add a comment to remind future people to check if newer version of upstream have added completion for [other shells](https://github.com/nixos/nixpkgs/blob/1d3edd77efd72a36ce2508d3ba19de3c35452483/pkgs/build-support/setup-hooks/install-shell-files.sh#L81)?

**Review by ckiee — CHANGES_REQUESTED** [review]

**Review by tomberek — APPROVED** [review]
Minor fixes in previous reviews. Otherwise looks pretty good. Welcome to nixpkgs!

File: pkgs/top-level/all-packages.nix
Comment (lockejan) [reply pkgs/top-level/all-packages.nix:26144]: Hmm, okay, there are 99 occurrences of {} in this file already. Is there a style guide available that covers this? I mean, I don't know why to choose one over the other yet.

Comment (ckiee) [reply pkgs/top-level/all-packages.nix:26144]: > Hmm, okay, there are 99 occurrences of {} in this file already

And a *lot* more of `{ }`

> Is there a style guide available that covers this?

- Don't think so, seems it just gets passed down with reviews like this one. 
- You can use `nixpkgs-fmt` for newer files.

Comment (lockejan) [reply pkgs/top-level/all-packages.nix:26144]: Corrected it. Actually I'm already using nixpkgs-fmt. It's probably a good idea to run it over the whole file and put that in a .git-blame-ignore-revs file

Comment (ckiee) [reply pkgs/top-level/all-packages.nix:26144]: I'd agree if we didn't have at least 878 [open PRs for new packages](https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+is%3Apr+label%3A%228.has%3A+package+%28new%29%22). Merge conflicts would not be fun.

File: pkgs/applications/version-management/git-and-tools/git-team/default.nix
Comment (lockejan) [reply pkgs/applications/version-management/git-and-tools/git-team/default.nix:23]: I work in the same company as git-team's author and there's no plan to add more completions for years now. I'd rather call for yagni, but I added a comment for clarification anyways.

**Review by ckiee — APPROVED** [review]
Looks good. You can dismiss @zowoq's review at the bottom of this PR page to make it mergeable.

**Comment by zowoq** [comment]
Test failure on `x86_64-darwin`?

File: pkgs/applications/version-management/git-and-tools/git-team/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/version-management/git-and-tools/git-team/default.nix:28]:
```suggestion
    description = "Command line interface for managing and enhancing git commit messages with co-authors";
```

File: pkgs/applications/version-management/git-and-tools/git-team/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/version-management/git-and-tools/git-team/default.nix:19]:
```suggestion
    installManPage <(go run main.go --generate-man-page)
```

File: pkgs/applications/version-management/git-and-tools/git-team/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/version-management/git-and-tools/git-team/default.nix:23]:
```suggestion
    go run main.go completion bash > $out/${pname}.bash
    installShellCompletion --cmd git-team --bash <($out/bin/git-team completion bash)
```

File: pkgs/applications/version-management/git-and-tools/git-team/default.nix
Comment (lockejan) [reply pkgs/applications/version-management/git-and-tools/git-team/default.nix:28]: Nixfmt breaks it into two lines because it succeeds 80 characters.

File: pkgs/applications/version-management/git-and-tools/git-team/default.nix
Comment (lockejan) [reply pkgs/applications/version-management/git-and-tools/git-team/default.nix:19]: While I generally like the idea, this is actually failing on my machine during build.

>@nix { "action": "setPhase", "phase": "installPhase" }
installing
installManPage: error: path missing manpage section suffix: /dev/fd/63

File: pkgs/applications/version-management/git-and-tools/git-team/default.nix
Comment (lockejan) [reply pkgs/applications/version-management/git-and-tools/git-team/default.nix:23]: You probably also meant rm line 22, right? Otherwise it doesn't make sense to me.

**Comment by lockejan** [comment]
> Test failure on `x86_64-darwin`?
> 
> @ofborg build git-team

The tests are running successful locally under x86_64-darwin (with nix-update). Feels like a false positive.

File: pkgs/applications/version-management/git-and-tools/git-team/default.nix
Comment (lockejan) [reply pkgs/applications/version-management/git-and-tools/git-team/default.nix]: updated

Comment (ckiee) [reply pkgs/applications/version-management/git-and-tools/git-team/default.nix]: Sandro can be very picky :P

**Comment by cole-h** [comment]
@ofborg eval

(GitHub API badness :)

File: pkgs/applications/version-management/git-and-tools/git-team/default.nix
Comment (SuperSandro2000) [reply pkgs/applications/version-management/git-and-tools/git-team/default.nix]: > Nixfmt breaks it into two lines because it succeeds 80 characters

Yeah thats stupid because we don't enforce the 80 line limit. Just not long enough is fine.

> Sandro can be very picky :P

description should always be a single line and not using multiline strings makes sure that it is obvious when it is to long and should be moved to longDescription.

File: pkgs/applications/version-management/git-and-tools/git-team/default.nix
Comment (SuperSandro2000) [reply pkgs/applications/version-management/git-and-tools/git-team/default.nix:19]: 🤔 Then it might not work with installManPage. I am not sure actually. Anyway we don't want to pipe it to $out/*.1.

```suggestion
    go run main.go --generate-man-page > ${pname}.1
    installManPage ${pname}.1
```

File: pkgs/applications/version-management/git-and-tools/git-team/default.nix
Comment (SuperSandro2000) [reply pkgs/applications/version-management/git-and-tools/git-team/default.nix]: > You probably also meant rm line 22, right? Otherwise it doesn't make sense to me.

Yeah.

We don't want to create anything temporary under `$out/`. Also install shell completion like that is the nice and short way and also allows easily adding fish and zsh later.

File: pkgs/applications/version-management/git-and-tools/git-team/default.nix
Comment (lockejan) [reply pkgs/applications/version-management/git-and-tools/git-team/default.nix]: Updated

File: pkgs/applications/version-management/git-and-tools/git-team/default.nix
Comment (lockejan) [reply pkgs/applications/version-management/git-and-tools/git-team/default.nix]: Fine. I don't mind. Updated.
Where can I find more about the semantic regarding $out ?
Because I can't follow your argument why putting it in $out is a no-no.

Comment (SuperSandro2000) [reply pkgs/applications/version-management/git-and-tools/git-team/default.nix]: `$out` is basically what later will be the package contents. If you place temporary files there and leave them then will end up in the final package.

**Pull request merged**
